defaults/jq/semconv.jq

```diff
@@ -0,0 +1,102 @@
+def semconv_group_attributes_by_namespace:
+    group_by(.namespace)
+    | map({ namespace: .[0].namespace, attributes: . | sort_by(.name) });
+
+#####################
+# Attribute functions
+#####################
+def semconv_attributes($options):
+    .groups
+    | map(select(.type == "attribute_group" and (.id | startswith("registry."))))
+    | map(.attributes) | add
+    | if ($options | has("exclude_stability")) then
+        map(select(.stability as $st | $options.exclude_stability | index($st) | not))
+      else
+        .
+      end
+    | if ($options | has("exclude_deprecated") and $options.exclude_deprecated == true) then
+        map(select(has("deprecated") | not))
+      else
+        .
+      end
+    | map(. + {namespace: (if .name | index(".") then .name | split(".")[0] else "other" end)})
+    | if ($options | has("exclude_namespace")) then
+        map(select(.namespace as $st | $options.exclude_namespace | index($st) | not))
+      else
+        .
+      end
+    | sort_by(.namespace, .name);
```

Comment:
My main complain here is around `.namespace` here and elsewhere.

We are doing this today in codegen and semconv.  However, should namespace be the FIRST prefix before `.` or ALL BUT LAST before `.`.   How do we want to treat namespaces overall?

In the hopes of making things better, I think we go with this. It should be easy to clean up and modify going forward (as this is the cleanest JQ I've seen), but I *do* think we need to make hard decisions in semconv tooling about namespacing and what it means.

Reply:
we currently care about the root namespace in the codegen, but I agree that we'll probably need more flexibility in the future. 

For now I'd only propose to make it explicit in the naming (if `.namespace` here is the root one, then let's call it `root_namespace` or similar)

Reply:
I can rename `namespace` to `root_namespace` as I believe it aligns well with @lmolkova's definition. @jsuereth, do you agree with that?

Reply:
I think `root_namespace` is a fair compromise between making weaver usable and not creating concepts that are shifting sands and will break over time.

+1 to `root_namespace`

Reply:
Done